AttitudeManager/Src/AM.cpp

```diff
@@ -35,13 +35,93 @@ AttitudeManagerInput AttitudeManager::getControlInputs() {
     return temp;
 }
 
+AttitudeManager::AttitudeManager(Flightmode* control_algorithm):
+    control_algorithm_(control_algorithm),
+    motorInstances_(new MotorInstance_t[config::NUM_MOTORS]{})
```

Comment:
As I'm going through this PR, I'm feeling more and more like we should be using dependency injection to pass in the motors (the same way `control_algorithm` is injected. This allows you to inject mock motors in unit tests without needing to friend the class. This design pattern is why we have a parent `MotorChannel` and each driver inherits from it.

Reply:
That makes a lot of sense I'll make a commit with that soon

Reply:
I would still need to friend the class though as that is how I find out what the motors are being set to, I could fix this problem and [this problem](https://github.com/UWARG/ZeroPilot-SW-3.5/pull/31/files#r1367636496) at the same time by creating a global array that holds motor outputs (kinda simulating how the stm sets regkeys for motor outputs is my thinking)

Reply:
Actually after trying the global array way, it seems like I would still need to friend the class as outputToMotor is a private method that AM is supposed to call on its own so I don't know how else I could test it

Reply:
> I would still need to friend the class though as that is how I find out what the motors are being set to, I could fix this problem and [this problem](https://github.com/UWARG/ZeroPilot-SW-3.5/pull/31/files#r1367636496) at the same time by creating a global array that holds motor outputs (kinda simulating how the stm sets registers for motor outputs is my thinking)

You should not need to friend AM to inspect the value that a motor was set to, as the test will have the motor channel in its local scope to inspect.